### PR TITLE
Docstring for `isnumeric`

### PR DESCRIPTION
**Pull request by mcabbott**

### CODE REVIEW

File: docs/src/api.md [hunk 1]
```diff
@@ -37,11 +37,14 @@ Optimisers.update!
 Optimisers.adjust(::Any, ::Real)
 ```
 
-Calling `Functors.@functor` on your model's layer types by default causes the
-optimiser to act on all suitable fields. To restrict this, define `trainable`:
+Calling `Functors.@functor` on your model's layer types by default causes
+these functions to recurse into all children, and ultimately optimise
+all `isnumeric` leaf nodes.
```
Comment (ToucheSir) [inline docs/src/api.md:42]: Would adding a ref link work here?

**Comment by Saransh-cpp** [comment]
The preview - https://fluxml.ai/Optimisers.jl/previews/PR104/api/#Optimisers.isnumeric

File: docs/src/api.md (hunk 1 above)
Comment (Saransh-cpp) [reply docs/src/api.md:42]: Should work!

Comment (mcabbott) [reply docs/src/api.md:42]: I guess my feeling is that these should not be links. This sentence is introducing the docstrings immediately below, and a link that scrolls down 2cm seems more likely to be confusing than helpful.

Comment (Saransh-cpp) [reply docs/src/api.md:42]: Oh yes, I overlooked that.

**Review by ToucheSir — APPROVED** [review]

**Pull request merged**